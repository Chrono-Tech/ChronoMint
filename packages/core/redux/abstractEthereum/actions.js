/**
 * Copyright 2017â€“2019, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import {
  NONCE_UPDATE,
} from './constants'

export const nonceUpdate = (blockchain) => (address, nonce) => ({
  type: NONCE_UPDATE,
  address,
  blockchain,
  nonce,
})
