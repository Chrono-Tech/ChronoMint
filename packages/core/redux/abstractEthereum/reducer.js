/**
 * Copyright 2017â€“2019, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import {
  NONCE_UPDATE,
} from './constants'

import {
  BLOCKCHAIN_ETHEREUM,
  BLOCKCHAIN_LABOR_HOUR,
} from '../../dao/constants'

const initialSubState = {
  nonces: {},
}

const initialState = () => ({
  [BLOCKCHAIN_ETHEREUM]: {
    ...initialSubState,
  },
  [BLOCKCHAIN_LABOR_HOUR]: {
    ...initialSubState,
  },
})

const mutations = {
  [NONCE_UPDATE]: (state, { address, blockchain, nonce }) => {
    const blockchainScope = state[blockchain]
    const nonces = blockchainScope.nonces

    return ({
      ...state,
      [blockchain]: {
        ...blockchainScope,
        nonces: {
          ...nonces,
          [address]: nonce,
        },
      },
    })
  },
}

export default (state = initialState(), { type, ...payload }) => {
  return (type in mutations)
    ? mutations[type](state, payload)
    : state
}
