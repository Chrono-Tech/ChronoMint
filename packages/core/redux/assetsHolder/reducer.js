/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import AssetHolderModel from '../../models/assetHolder/AssetHolderModel'
import * as a from './constants'

const initialState = new AssetHolderModel()

export default (state = initialState, action) => {
  switch (action.type) {
    case a.ASSET_HOLDER_INIT:
      return state.isInited(true)
    case a.ASSET_HOLDER_ADDRESS:
      return state
        .account(action.account)
        .wallet(action.wallet)
    case a.ASSET_HOLDER_ASSET_UPDATE:
      return state.assets(state.assets().update(action.asset))
    default:
      return state
  }
}
