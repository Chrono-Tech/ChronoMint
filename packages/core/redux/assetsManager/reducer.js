/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import MainAssetsManagerModel from '../../models/assetsManager/MainAssetsManagerModel'
import {
  GET_ASSETS_MANAGER_COUNTS,
  GET_ASSETS_MANAGER_COUNTS_START,
  GET_PLATFORMS,
  GET_TRANSACTIONS_DONE,
  GET_TRANSACTIONS_START,
  SELECT_PLATFORM,
  SELECT_TOKEN,
  SET_ASSETS,
  SET_NEW_MANAGERS_LIST,
  GET_ASSET_DATA,
} from './constants'

export const initialState = new MainAssetsManagerModel()

export default (state = initialState, action) => {
  switch (action.type) {
    case GET_ASSETS_MANAGER_COUNTS_START:
      return state.isFetching(true).isFetched(false)
    case GET_ASSETS_MANAGER_COUNTS:
      return state
        .isFetching(false)
        .isFetched(true)
        .managersCount(action.payload.managers.length || 0)
        .assets(action.payload.assets)
        .platformsList(action.payload.platforms)
        .usersPlatforms(action.payload.usersPlatforms)
    case GET_ASSET_DATA:
      return state
        .assets({
          ...state.assets(),
          [ action.asset.token ]: action.asset,
        })
    case GET_PLATFORMS:
      return state
        .platformsList(action.payload.platforms)
        .usersPlatforms(action.payload.usersPlatforms)
    case SELECT_TOKEN:
      return state.selectedToken(action.payload.symbol)
    case SELECT_PLATFORM:
      return state.selectedPlatform(action.payload.platformAddress)
    case SET_ASSETS:
      return state.assets(action.payload.assets)
    case SET_NEW_MANAGERS_LIST:
      return state.managersCount(action.payload.managers.length)
    case GET_TRANSACTIONS_START:
      return state
        .transactionsList(
          state.transactionsList()
            .isFetched(false)
            .isFetching(true),
        )
    case GET_TRANSACTIONS_DONE:
      return state
        .transactionsList(
          state.transactionsList()
            .merge(action.payload.transactionsList)
            .isFetched(true)
            .isFetching(false),
        )
    default:
      return state
  }
}
