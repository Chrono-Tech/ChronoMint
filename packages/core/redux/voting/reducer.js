/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import VotingMainModel from '../../models/voting/VotingMainModel'
import {
  POLLS_CREATE,
  POLLS_LIST,
  POLLS_LOAD,
  POLLS_REMOVE,
  POLLS_SELECTED,
  POLLS_UPDATE,
  POLLS_VOTE_LIMIT,
} from './constants'

const initialState = new VotingMainModel()

export default (state = initialState, action) => {
  switch (action.type) {
    case POLLS_VOTE_LIMIT:
      return state
        .voteLimitInTIME(action.voteLimitInTIME)
        .voteLimitInPercent(action.voteLimitInPercent)
    case POLLS_LOAD:
      return state.isFetching(true)
    case POLLS_LIST:
      return state
        .isFetching(false)
        .isFetched(true)
        .list(action.list)
    case POLLS_CREATE:
      return state
        .list(state.list().update(action.poll))
    case POLLS_UPDATE:
      return state.list(state.list().update(action.poll))
    case POLLS_REMOVE:
      return state
        .list(state.list().remove(action.id))
    case POLLS_SELECTED:
      return state.selectedPoll(action.id)
    default:
      return state
  }
}
