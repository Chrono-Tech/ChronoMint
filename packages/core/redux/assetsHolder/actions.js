/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import Amount from '../../models/Amount'
import AssetModel from '../../models/assetHolder/AssetModel'
import TokenModel from '../../models/tokens/TokenModel'
import AllowanceModel from '../../models/wallet/AllowanceModel'
import { DUCK_SESSION } from '../session/constants'
import { subscribeOnTokens } from '../tokens/actions'
import tokenService from '../../services/TokenService'
import { daoByType } from '../daos/selectors'
import TxExecModel from '../../models/TxExecModel'
import { getWallet } from '../wallets/selectors/models'
import { WALLETS_UPDATE_WALLET } from '../wallets/constants'
import WalletModel from '../../models/wallet/WalletModel'
import AllowanceCollection from '../../models/AllowanceCollection'

import {
  DUCK_ASSETS_HOLDER,
  ASSET_HOLDER_INIT,
  ASSET_HOLDER_ADDRESS,
  ASSET_HOLDER_ASSET_UPDATE,
} from './constants'

const handleToken = (token: TokenModel) => async (dispatch, getState) => {
  const assetHolder = getState().get(DUCK_ASSETS_HOLDER)
  const assets = assetHolder.assets()

  if (!token.isERC20() || !assets.list().has(token.address())) {
    return
  }

  const holderWallet = assetHolder.wallet()

  // set symbol for asset
  const asset = assets.item(token.address()).symbol(token.symbol())
  dispatch({ type: ASSET_HOLDER_ASSET_UPDATE, asset })

  // subscribe to token
  const tokenDAO = tokenService.getDAO(token.id())
  tokenDAO
    .on('approval', (results) => {
      if (results.from === holderWallet || results.spender === holderWallet) {
        dispatch(fetchAssetAllowance(token))
      }
    })
    .on('transfer', (data) => {
      if (!(data.from === holderWallet || data.to === holderWallet)) {
        return
      }
      dispatch(fetchAssetDeposit(token))
      dispatch(fetchAssetAllowance(token))
    })

  // need to be uncomment
  // const holderAccount = assetHolder.account()
  // await tokenDAO.watch(holderAccount)

  // fetch deposit and allowance
  dispatch(fetchAssetDeposit(token))
  dispatch(fetchAssetAllowance(token))
}

export const fetchAssetDeposit = (token: TokenModel) => async (dispatch, getState) => {
  const { account } = getState().get(DUCK_SESSION)
  const assetHolderDAO = daoByType('TimeHolder')(getState())
  const deposit = await assetHolderDAO.getDeposit(token.address(), account)
  const asset = getState().get(DUCK_ASSETS_HOLDER).assets().item(token.address()).deposit(new Amount(
    deposit,
    token.symbol(),
  ))
  dispatch({ type: ASSET_HOLDER_ASSET_UPDATE, asset })
}

export const fetchAssetAllowance = (token: TokenModel) => async (dispatch, getState) => {
  const assetHolder = getState().get(DUCK_ASSETS_HOLDER)
  const { account } = getState().get(DUCK_SESSION)

  const holderWallet = assetHolder.wallet()
  const tokenDAO = tokenService.getDAO(token.id())
  const assetHolderWalletAllowance = await tokenDAO.getAccountAllowance(account, holderWallet)

  const wallet = getWallet(`Ethereum-${account}`)(getState())
  const allowance = new AllowanceModel({
    amount: new Amount(assetHolderWalletAllowance, token.id()),
    spender: holderWallet,
    token: token.id(),
    isFetching: false,
    isFetched: true,
  })

  dispatch({
    type: WALLETS_UPDATE_WALLET,
    wallet: new WalletModel({
      ...wallet,
      allowances: new AllowanceCollection({
        list: {
          ...wallet.allowances.list,
          [allowance.id()]: allowance,
        },
      }),
    }),
  })
}

export const initAssetsHolder = () => async (dispatch, getState) => {
  if (getState().get(DUCK_ASSETS_HOLDER).isInited()) {
    return
  }
  dispatch({ type: ASSET_HOLDER_INIT, inInited: true })

  const assetHolderDAO = daoByType('TimeHolder')(getState())
  const [walletAddress] = await Promise.all([
    assetHolderDAO.getWalletAddress(),
  ])

  dispatch({ type: ASSET_HOLDER_ADDRESS, account: assetHolderDAO.address, wallet: walletAddress.toLowerCase() })

  // get assets list
  const [timeAddress] = await Promise.all([
    assetHolderDAO.getSharesContract(),
  ])
  const assets = [timeAddress]
  assets.forEach((address) => {
    dispatch({
      type: ASSET_HOLDER_ASSET_UPDATE,
      asset: new AssetModel({
        address: address.toLowerCase(),
      }),
    })
  })

  dispatch(subscribeOnTokens(handleToken))
}

export const depositAsset = (amount: Amount, token: TokenModel, feeMultiplier: Number = 1, advancedOptions = undefined) => async (dispatch, getState) => {
  try {
    const assetHolderDAO = daoByType('TimeHolder')(getState())
    const { account } = getState().get(DUCK_SESSION)
    advancedOptions['account'] = account
    await assetHolderDAO.deposit(token, amount, feeMultiplier, advancedOptions)
  } catch (e) {
    // eslint-disable-next-line
    console.error('deposit error', e)
  }
}

export const withdrawAsset = (amount: Amount, token: TokenModel, feeMultiplier: Number = 1, advancedOptions = undefined) => async (dispatch, getState) => {
  try {
    const assetHolderDAO = daoByType('TimeHolder')(getState())
    const { account } = getState().get(DUCK_SESSION)
    advancedOptions['account'] = account
    const tx: TxExecModel = await assetHolderDAO.withdraw(token, amount, feeMultiplier, advancedOptions)
  } catch (e) {
    // eslint-disable-next-line
    console.error('withdraw error', e)
  }
}
