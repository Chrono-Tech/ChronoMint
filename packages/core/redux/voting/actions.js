/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import BigNumber from 'bignumber.js'
import { push } from '@chronobank/core-dependencies/router'
import Amount from '../../models/Amount'
import { ETH } from '../../redux/mainWallet/actions'
import { DUCK_SESSION } from '../session/actions'
import votingService from '../../services/VotingService'
import contractsManagerDAO from '../../dao/ContractsManagerDAO'
import { EVENT_POLL_ACTIVATED, EVENT_POLL_ENDED, EVENT_POLL_VOTED } from '../../dao/PollEmitterDAO'
import type PollNoticeModel from '../../models/notices/PollNoticeModel'
import { IS_ACTIVATED, IS_CREATED, IS_ENDED, IS_REMOVED, IS_UPDATED, IS_VOTED } from '../../models/notices/PollNoticeModel'
import PollDetailsModel from '../../models/PollDetailsModel'
import { notify } from '../notifier/actions'
import { EVENT_POLL_CREATED, EVENT_POLL_REMOVED, TX_CREATE_POLL } from '../../dao/VotingManagerDAO'
import { PTPoll } from './types'
import { getSelectedPollFromDuck, getVoting } from './selectors/models'
import { daoByType } from '../../refactor/redux/daos/selectors'
import { sendNewTx } from '../../refactor/redux/transactions/actions'

export const POLLS_VOTE_LIMIT = 'voting/POLLS_LIMIT'
export const POLLS_LOAD = 'voting/POLLS_LOAD'
export const POLLS_LIST = 'voting/POLLS_LIST'
export const POLLS_CREATE = 'voting/POLLS_CREATE'
export const POLLS_REMOVE = 'voting/POLLS_REMOVE'
export const POLLS_UPDATE = 'voting/POLLS_UPDATE'
export const POLLS_SELECTED = 'voting/POLLS_SELECTED'

export const DUCK_VOTING = 'voting'
const PAGE_SIZE = 20

// used to create unique ID for fetching models
let counter = 1

export const goToVoting = () => (dispatch) => dispatch(push('/voting'))

export const watchPoll = (notice: PollNoticeModel) => async (dispatch) => {
  switch (notice.status()) {
    case IS_CREATED:
      dispatch(handlePollRemoved(notice.transactionHash()))
      dispatch(handlePollCreated(notice.poll()))
      break
    case IS_REMOVED:
      dispatch(handlePollRemoved(notice.pollId()))
      break
    case IS_ACTIVATED:
      dispatch(handlePollUpdated(notice.poll()))
      break
    case IS_ENDED:
      dispatch(handlePollUpdated(notice.poll()))
      break
    case IS_VOTED:
    case IS_UPDATED:
      dispatch(handlePollUpdated(notice.poll()))
      break
  }
  dispatch(notify(notice))
}

export const updateVoteLimit = () => async (dispatch, getState) => {
  // const votingDAO = await contractsManagerDAO.getVotingManagerDAO()
  const votingDAO = daoByType('VotingManager')(getState())
  const [voteLimitInTIME, voteLimitInPercent] = await Promise.all([
    votingDAO.getVoteLimit(),
    votingDAO.getVoteLimitInPercent(),
  ])
  dispatch({ type: POLLS_VOTE_LIMIT, voteLimitInTIME, voteLimitInPercent: new BigNumber(voteLimitInPercent).div(100) })
}

export const watchInitPolls = () => async (dispatch, getState) => {
  const callback = (notice) => dispatch(watchPoll(notice))
  const { account } = getState().get(DUCK_SESSION)
  const votingManagerDAO = daoByType('VotingManager')(getState())
  votingService
    .setVotingManager(votingManagerDAO)
  votingService
    .subscribeToVoting(account)

  votingService
    .on(EVENT_POLL_CREATED, callback)
    .on(EVENT_POLL_REMOVED, callback)
    .on(EVENT_POLL_ACTIVATED, callback)
    .on(EVENT_POLL_ENDED, callback)
    .on(EVENT_POLL_VOTED, callback)

  return Promise.all([
    dispatch(updateVoteLimit()),
  ])
}

export const createPoll = (poll: PollDetailsModel) => async (dispatch, getState) => {
  const { account } = getState().get(DUCK_SESSION)
  const id = `stub_${--counter}`
  const stub = poll.mutate({ id: id, isFetching: true })
  const votingDAO = daoByType('VotingManager')(getState())

  try {
    dispatch(handlePollCreated(stub))
    dispatch(goToVoting())
    const tx = await votingDAO.createPoll(poll.poll, { from: account })
    dispatch(sendNewTx(tx, (transactionHash) => {
      dispatch(handlePollRemoved(stub.id))
      dispatch(handlePollUpdated(stub.mutate({ transactionHash: transactionHash })))
    }))

  } catch (e) {
    dispatch(handlePollRemoved(stub.id))
  }
}

export const removePoll = (pollObject: PTPoll) => async (dispatch, getState) => {
  const state = getState()
  const { account } = state.get(DUCK_SESSION)
  const votingDAO = daoByType('VotingManager')(getState())

  let poll = pollObject && pollObject.id
    ? getVoting(state).list().item(pollObject.id)
    : getSelectedPollFromDuck(state)

  try {
    dispatch(handlePollRemoved(poll.id))
    dispatch(goToVoting())
    const dao = await votingDAO.pollInterfaceManagerDAO.getPollInterfaceDAO(poll.id)

    const tx = await dao.removePoll({ from: account, symbol: 'TIME' })
    dispatch(sendNewTx(tx))
  } catch (e) {
    dispatch(handlePollCreated(poll))
    throw e
  }
}

export const vote = (choice: Number) => async (dispatch, getState) => {
  const poll = getSelectedPollFromDuck(getState())
  const { account } = getState().get(DUCK_SESSION)
  const votingDAO = daoByType('VotingManager')(getState())

  try {
    dispatch(handlePollUpdated(poll.isFetching(true)))
    const options = poll.voteEntries()
    const dao = await votingDAO.pollInterfaceManagerDAO.getPollInterfaceDAO(poll.id)
    const tx = await dao.vote(choice, options.get(choice), { from: account, symbol: 'TIME' })
    dispatch(sendNewTx(tx))
  } catch (e) {
    dispatch(handlePollUpdated(poll))
    throw e
  }
}

export const activatePoll = (pollObject: PTPoll) => async (dispatch, getState) => {
  const state = getState()
  const { account } = getState().get(DUCK_SESSION)
  const votingDAO = daoByType('VotingManager')(getState())

  let poll = pollObject && pollObject.id
    ? getVoting(state).list().item(pollObject.id)
    : getSelectedPollFromDuck(state)

  try {
    dispatch(handlePollUpdated(poll
      .poll(poll.poll.mutate({ active: true, isFetching: true }))))

    const dao = await votingDAO.pollInterfaceManagerDAO.getPollInterfaceDAO(poll.id)
    await dao.activatePoll({ from: account, symbol: 'TIME' })
  } catch (e) {
    dispatch(handlePollUpdated(poll))
  }
}

export const endPoll = (pollObject: PTPoll) => async (dispatch, getState) => {
  const poll = getState().get(DUCK_VOTING).list().item(pollObject.id)
  const { account } = getState().get(DUCK_SESSION)
  const votingDAO = daoByType('VotingManager')(getState())

  try {
    dispatch(handlePollUpdated(poll
      .set('poll', poll.poll.mutate({
        active: false,
        status: false,
        isFetching: true,
      }))))
    const dao = await votingDAO.pollInterfaceManagerDAO.getPollInterfaceDAO(poll.id)
    await dao.endPoll({ from: account, symbol: 'TIME' })
  } catch (e) {
    dispatch(handlePollUpdated(poll))
  }
}

export const handlePollCreated = (poll: PollDetailsModel) => async (dispatch) => {
  poll && dispatch({ type: POLLS_CREATE, poll })
}

export const handlePollRemoved = (id: Number) => async (dispatch) => {
  dispatch({ type: POLLS_REMOVE, id })
}

export const handlePollUpdated = (poll: PollDetailsModel) => async (dispatch) => {
  dispatch({ type: POLLS_UPDATE, poll })
}

export const listPolls = () => async (dispatch) => {
  dispatch({ type: POLLS_LOAD })
  const list = await dispatch(getNextPage())
  dispatch({ type: POLLS_LIST, list })
}

export const getNextPage = () => async (dispatch, getState) => {
  const dao = daoByType('VotingManager')(getState())

  const votingState = getState().get(DUCK_VOTING)
  const { account } = getState().get(DUCK_SESSION)
  return dao.getPollsPaginated(votingState.lastPoll(), PAGE_SIZE, account)
}

export const estimateGasForVoting = async (mode: string, params, callback, gasPriceMultiplier = 1) => {
  let dao = null
  switch (mode) {
    case TX_CREATE_POLL:
      dao = await contractsManagerDAO.getVotingManagerDAO()
      break
  }
  try {
    if (!dao) {
      callback(new Error('Dao is undefined'))
    }
    const { gasLimit, gasFee, gasPrice } = await dao.estimateGas(...params)
    callback(null, {
      gasLimit,
      gasFee: new Amount(gasFee.mul(gasPriceMultiplier), ETH),
      gasPrice: new Amount(gasPrice.mul(gasPriceMultiplier), ETH),
    })
  } catch (e) {
    callback(e)
  }
}
