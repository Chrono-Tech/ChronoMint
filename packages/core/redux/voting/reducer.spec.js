/**
 * Copyright 2017â€“2018, LaborX PTY
 * Licensed under the AGPL Version 3 license.
 */

import BigNumber from 'bignumber.js'
import VotingCollection from '../../models/voting/VotingCollection'
import PollModel from '../../models/PollModel'
import PollDetailsModel from '../../models/PollDetailsModel'
import VotingMainModel from '../../models/voting/VotingMainModel'
import * as a from './actions'
import reducer from './reducer'

const initialState = new VotingMainModel()
const poll = new PollDetailsModel({
  poll: new PollModel()
    .set('id', 1)
    .set('published', null)
    .set('deadline', null),
})
let votingsList = new VotingCollection()
votingsList = votingsList.add(poll)

describe('assetsManager reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })

  it('sould POLLS_VOTE_LIMIT', () => {
    expect(
      reducer(undefined, {
        type: a.POLLS_VOTE_LIMIT,
        voteLimitInTIME: new BigNumber(1),
        voteLimitInPercent: new BigNumber(1),
      }),
    ).toMatchSnapshot()
  })

  it('sould handle POLLS_LOAD', () => {
    expect(
      reducer(undefined, {
        type: a.POLLS_LOAD,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle POLLS_LIST', () => {
    expect(
      reducer(undefined, {
        type: a.POLLS_LIST,
        list: votingsList,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle POLLS_CREATE', () => {
    expect(
      reducer(undefined, {
        type: a.POLLS_CREATE,
        poll,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle POLLS_UPDATE', () => {
    expect(
      reducer(undefined, {
        type: a.POLLS_UPDATE,
        poll,
      }),
    ).toMatchSnapshot()
  })

  it('sould handle POLLS_REMOVE', () => {
    const state = initialState.list(votingsList)
    expect(
      reducer(state, {
        type: a.POLLS_REMOVE,
        id: '1',
      }),
    ).toMatchSnapshot()
  })

})
