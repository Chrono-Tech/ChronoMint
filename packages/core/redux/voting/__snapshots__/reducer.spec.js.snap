// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assetsManager reducer sould POLLS_VOTE_LIMIT 1`] = `
{
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isInited": false,
  "isPending": false,
  "isSelected": false,
  "lastPoll": "0",
  "list": {
    "emptyModel": null,
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isInited": false,
    "isPending": false,
    "isSelected": false,
    "leftToFetch": 0,
    "list": {},
    "selected": null,
    "timestamp": null,
    "transactionHash": null
  },
  "timestamp": null,
  "transactionHash": null,
  "voteLimitInPercent": "1",
  "voteLimitInTIME": "1"
}
`;

exports[`assetsManager reducer sould handle POLLS_CREATE 1`] = `
{
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isInited": false,
  "isPending": false,
  "isSelected": false,
  "lastPoll": "0",
  "list": {
    "emptyModel": null,
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isInited": false,
    "isPending": false,
    "isSelected": false,
    "leftToFetch": 0,
    "list": {
      "null": {
        "files": [],
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isInited": false,
        "isPending": false,
        "isSelected": false,
        "poll": {
          "active": false,
          "deadline": null,
          "description": "",
          "files": null,
          "hasMember": false,
          "hash": null,
          "id": 1,
          "isFailed": false,
          "isFetched": false,
          "isFetching": false,
          "isInited": false,
          "isPending": false,
          "isSelected": false,
          "isTransaction": false,
          "memberOption": null,
          "options": [
            "Support",
            "Decline"
          ],
          "owner": null,
          "published": null,
          "status": false,
          "timestamp": null,
          "title": "",
          "transactionHash": null,
          "voteLimitInTIME": null
        },
        "shareholdersCount": "0",
        "statistics": [],
        "timestamp": null,
        "totalSupply": "0",
        "transactionHash": null,
        "votes": []
      }
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null
  },
  "timestamp": null,
  "transactionHash": null,
  "voteLimitInPercent": null,
  "voteLimitInTIME": null
}
`;

exports[`assetsManager reducer sould handle POLLS_LIST 1`] = `
{
  "id": null,
  "isFailed": false,
  "isFetched": true,
  "isFetching": false,
  "isInited": false,
  "isPending": false,
  "isSelected": false,
  "lastPoll": "0",
  "list": {
    "emptyModel": null,
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isInited": false,
    "isPending": false,
    "isSelected": false,
    "leftToFetch": 0,
    "list": {
      "null": {
        "files": [],
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isInited": false,
        "isPending": false,
        "isSelected": false,
        "poll": {
          "active": false,
          "deadline": null,
          "description": "",
          "files": null,
          "hasMember": false,
          "hash": null,
          "id": 1,
          "isFailed": false,
          "isFetched": false,
          "isFetching": false,
          "isInited": false,
          "isPending": false,
          "isSelected": false,
          "isTransaction": false,
          "memberOption": null,
          "options": [
            "Support",
            "Decline"
          ],
          "owner": null,
          "published": null,
          "status": false,
          "timestamp": null,
          "title": "",
          "transactionHash": null,
          "voteLimitInTIME": null
        },
        "shareholdersCount": "0",
        "statistics": [],
        "timestamp": null,
        "totalSupply": "0",
        "transactionHash": null,
        "votes": []
      }
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null
  },
  "timestamp": null,
  "transactionHash": null,
  "voteLimitInPercent": null,
  "voteLimitInTIME": null
}
`;

exports[`assetsManager reducer sould handle POLLS_LOAD 1`] = `
{
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": true,
  "isInited": false,
  "isPending": false,
  "isSelected": false,
  "lastPoll": "0",
  "list": {
    "emptyModel": null,
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isInited": false,
    "isPending": false,
    "isSelected": false,
    "leftToFetch": 0,
    "list": {},
    "selected": null,
    "timestamp": null,
    "transactionHash": null
  },
  "timestamp": null,
  "transactionHash": null,
  "voteLimitInPercent": null,
  "voteLimitInTIME": null
}
`;

exports[`assetsManager reducer sould handle POLLS_REMOVE 1`] = `
{
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isInited": false,
  "isPending": false,
  "isSelected": false,
  "lastPoll": "0",
  "list": {
    "emptyModel": null,
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isInited": false,
    "isPending": false,
    "isSelected": false,
    "leftToFetch": 0,
    "list": {
      "null": {
        "files": [],
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isInited": false,
        "isPending": false,
        "isSelected": false,
        "poll": {
          "active": false,
          "deadline": null,
          "description": "",
          "files": null,
          "hasMember": false,
          "hash": null,
          "id": 1,
          "isFailed": false,
          "isFetched": false,
          "isFetching": false,
          "isInited": false,
          "isPending": false,
          "isSelected": false,
          "isTransaction": false,
          "memberOption": null,
          "options": [
            "Support",
            "Decline"
          ],
          "owner": null,
          "published": null,
          "status": false,
          "timestamp": null,
          "title": "",
          "transactionHash": null,
          "voteLimitInTIME": null
        },
        "shareholdersCount": "0",
        "statistics": [],
        "timestamp": null,
        "totalSupply": "0",
        "transactionHash": null,
        "votes": []
      }
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null
  },
  "timestamp": null,
  "transactionHash": null,
  "voteLimitInPercent": null,
  "voteLimitInTIME": null
}
`;

exports[`assetsManager reducer sould handle POLLS_UPDATE 1`] = `
{
  "id": null,
  "isFailed": false,
  "isFetched": false,
  "isFetching": false,
  "isInited": false,
  "isPending": false,
  "isSelected": false,
  "lastPoll": "0",
  "list": {
    "emptyModel": null,
    "id": null,
    "isFailed": false,
    "isFetched": false,
    "isFetching": false,
    "isInited": false,
    "isPending": false,
    "isSelected": false,
    "leftToFetch": 0,
    "list": {
      "null": {
        "files": [],
        "id": null,
        "isFailed": false,
        "isFetched": false,
        "isFetching": false,
        "isInited": false,
        "isPending": false,
        "isSelected": false,
        "poll": {
          "active": false,
          "deadline": null,
          "description": "",
          "files": null,
          "hasMember": false,
          "hash": null,
          "id": 1,
          "isFailed": false,
          "isFetched": false,
          "isFetching": false,
          "isInited": false,
          "isPending": false,
          "isSelected": false,
          "isTransaction": false,
          "memberOption": null,
          "options": [
            "Support",
            "Decline"
          ],
          "owner": null,
          "published": null,
          "status": false,
          "timestamp": null,
          "title": "",
          "transactionHash": null,
          "voteLimitInTIME": null
        },
        "shareholdersCount": "0",
        "statistics": [],
        "timestamp": null,
        "totalSupply": "0",
        "transactionHash": null,
        "votes": []
      }
    },
    "selected": null,
    "timestamp": null,
    "transactionHash": null
  },
  "timestamp": null,
  "transactionHash": null,
  "voteLimitInPercent": null,
  "voteLimitInTIME": null
}
`;
